central/role/resources/list.go

```diff
@@ -24,56 +24,92 @@ import (
 //
 // KEEP THE FOLLOWING LIST SORTED IN LEXICOGRAPHIC ORDER (case-sensitive).
 var (
-	APIToken                         = newResourceMetadata("APIToken", permissions.GlobalScope)
-	Alert                            = newResourceMetadata("Alert", permissions.NamespaceScope)
-	AllComments                      = newResourceMetadata("AllComments", permissions.GlobalScope)
-	AuthPlugin                       = newResourceMetadata("AuthPlugin", permissions.GlobalScope)
-	AuthProvider                     = newResourceMetadata("AuthProvider", permissions.GlobalScope)
-	BackupPlugins                    = newResourceMetadata("BackupPlugins", permissions.GlobalScope)
-	CVE                              = newResourceMetadata("CVE", permissions.NamespaceScope)
-	Cluster                          = newResourceMetadata("Cluster", permissions.ClusterScope)
-	Compliance                       = newResourceMetadata("Compliance", permissions.ClusterScope)
-	ComplianceRunSchedule            = newResourceMetadata("ComplianceRunSchedule", permissions.GlobalScope)
-	ComplianceRuns                   = newResourceMetadata("ComplianceRuns", permissions.ClusterScope)
-	Config                           = newResourceMetadata("Config", permissions.GlobalScope)
-	DebugLogs                        = newResourceMetadata("DebugLogs", permissions.GlobalScope)
-	Deployment                       = newResourceMetadata("Deployment", permissions.NamespaceScope)
-	Detection                        = newResourceMetadata("Detection", permissions.GlobalScope)
-	Group                            = newResourceMetadata("Group", permissions.GlobalScope)
-	Image                            = newResourceMetadata("Image", permissions.NamespaceScope)
-	ImageComponent                   = newResourceMetadata("ImageComponent", permissions.NamespaceScope)
-	ImageIntegration                 = newResourceMetadata("ImageIntegration", permissions.GlobalScope)
+	// Access is the new resource grouping all access related resources.
+	Access = newResourceMetadata("Access", permissions.GlobalScope)
+	// Administration is the new resource grouping all administration-like resources.
+	Administration = newResourceMetadata("Administration", permissions.GlobalScope)
```

Comment:
Right now created essentially a "resource replacement", which will take the original, old resource and potentially a new, grouped one. During access checks, access allowed will be enough for at least one of the resources. This will be done within the existing `sac.ForResourceHelper`. 

The issue I saw with `resourceGroupMetadata` was that it would evaluate against the full list of resources, i.e. given we want to guard access to `APIToken`, the `resourceGroupMetadata Integrations` would yield `Allowed` if a user had access to i.e. `SignatureIntegration`, which would break backwards compatibility.